vsphere/datadog_checks/vsphere/api_rest.py

```diff
@@ -3,29 +3,38 @@
 # Licensed under Simplified BSD License (see LICENSE)
 import json
 from collections import defaultdict
-from typing import Any, Dict, List
+from typing import Any, Dict, Iterator, List, Set
 
 from pyVmomi import vim
 
 from datadog_checks.base.log import CheckLoggingAdapter
 from datadog_checks.base.utils.http import RequestsWrapper
 from datadog_checks.vsphere.config import VSphereConfig
+from datadog_checks.vsphere.constants import ALL_RESOURCES_WITH_METRICS
 from datadog_checks.vsphere.types import ResourceTags, TagAssociation
 
-from .api import APIResponseError, smart_retry
+from .api import APIResponseError
 
-MOR_TYPE_MAPPING = {
+MOR_TYPE_MAPPING_FROM_STRING = {
     'HostSystem': vim.HostSystem,
     'VirtualMachine': vim.VirtualMachine,
     'Datacenter': vim.Datacenter,
     'Datastore': vim.Datastore,
     'ClusterComputeResource': vim.ClusterComputeResource,
 }
 
+MOR_TYPE_MAPPING_TO_STRING = {
+    vim.HostSystem: 'HostSystem',
+    vim.VirtualMachine: 'VirtualMachine',
+    vim.Datacenter: 'Datacenter',
+    vim.Datastore: 'Datastore',
+    vim.ClusterComputeResource: 'ClusterComputeResource',
+}
```

Comment:
I think it will make it easier to maintain if we build `MOR_TYPE_MAPPING_TO_STRING` using `MOR_TYPE_MAPPING_FROM_STRING`